Dev sentinel 2022 01 01 preview

anat-gilenson opened the pull request:
# Please note all PRs to the develop branch have already been reviewed and approved. 
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [X] SDK of .NET (need service team to ensure code readiness)
      - [X] SDK of Python
      - [X] SDK of Java
      - [X] SDK of Js
      - [X] SDK of Go
      - [X] PowerShell
      - [X] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

anat-gilenson committed:
Adds base for updating Microsoft.SecurityInsights from version preview/2021-10-01-preview to version 2022-01-01-preview

anat-gilenson committed:
Updates readme

anat-gilenson committed:
Updates API version in new specs and examples

debroy committed:
SourceControl API 2022-01-01-preview version - adding new attribute to SourceControl model (#17849)

* SourceControl 2022-01-01-preview version change

* fix validation error

* prettified

* prettified

Co-authored-by: debjyoti11 <debjyotiroy88@gmail.com>

roeybc committed:
New metadata API (#17942)

* metadata changes

* addressed changes

* updated enum kind

* updated values as well

Co-authored-by: Roey Ben Chaim <robencha@microsoft.com>

roeybc committed:
added new enum to kind (#17994)

Co-authored-by: Roey Ben Chaim <robencha@microsoft.com>

debroy committed:
Add flag to rotate webhook secret (#17989)

Co-authored-by: debjyoti11 <debjyotiroy88@gmail.com>

j5lim reviewed (approved):
Approved from ARM side.

laithhisham committed:
Align definitions with stable version (#17931)

* Align definitions with stable version

* align scheduled rule and template properties with stable

ushasan committed:
Note: We are creating a new api-version with this change. Watchlists.json 2022-01-01-preview parameter name changes (#17848)

* changed source to sourceType and watchlistAlias to alias

* Update CreateWatchlistAndWatchlistItems.json

* Update CreateWatchlistItem.json

* Update DeleteWatchlist.json

* Update DeleteWatchlistItem.json

* Update GetWatchlistByAlias.json

* Update GetWatchlistItemById.json

* Update GetWatchlistItems.json

* Update GetWatchlistByAlias.json

* added missing params to match stable version

* Revert "added missing params to match stable version"

This reverts commit a30a709613f94266d0eaba6495c92be43221a282.

Reverting by 1 commit

* refreshing pipeline

* reverting lint diff change

* attempting to fix RemovedPath breaking changes

* Revert "attempting to fix RemovedPath breaking changes"

This reverts commit 2f96ee3c050fde4c166bf730b0a1b5faff07cfc1.

reverting attempt at correcting breaking changes. cannot match preview
to stable without breaking everything.

* adding back source type for backwards compatibility, added response descriptions, corrected PR comments

* adding missing bracket

* ran npm run prettier-fix to correct style issues

* resolving model/definition errors

* Revert "resolving model/definition errors"

This reverts commit fd67a4f813d923965cefad50999010820446a107.

Reverting prettier changes made to other branches

* Revert "ran npm run prettier-fix to correct style issues"

This reverts commit 3055791e00e956fff3a62e24d8313dde96daa00c.

* update

* Update DeleteWatchlist.json

* Update Watchlists.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* fixing styling issues on Delete example

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* ran npm run prettier on DeleteWatchlist.json

* Update Watchlists.json

* reverting alias to watchlistAlias

Co-authored-by: Hassan Chawiche <hassanc@microsoft.com>

anat-gilenson committed:
Rename system to AutoAssigned (#18061)

* Rename system to AutoAssigned

* Add example to list operations

* Update API version in NRT examples

shschwar committed:
Execute validation tools and fix results (#17733)

* Execute validation tools and fic results

* Prettyfieir GetIoTById.json

ushasan committed:
Watchlists.json-2022-01-01-preview-parameterNameChanges (reverted PR reference for files changed -> https://github.com/Azure/azure-rest-api-specs/pull/17848/files) (#18085)

* changed source to sourceType and watchlistAlias to alias

* Update CreateWatchlistAndWatchlistItems.json

* Update CreateWatchlistItem.json

* Update DeleteWatchlist.json

* Update DeleteWatchlistItem.json

* Update GetWatchlistByAlias.json

* Update GetWatchlistItemById.json

* Update GetWatchlistItems.json

* Update GetWatchlistByAlias.json

* added missing params to match stable version

* Revert "added missing params to match stable version"

This reverts commit a30a709613f94266d0eaba6495c92be43221a282.

Reverting by 1 commit

* refreshing pipeline

* reverting lint diff change

* attempting to fix RemovedPath breaking changes

* Revert "attempting to fix RemovedPath breaking changes"

This reverts commit 2f96ee3c050fde4c166bf730b0a1b5faff07cfc1.

reverting attempt at correcting breaking changes. cannot match preview
to stable without breaking everything.

* adding back source type for backwards compatibility, added response descriptions, corrected PR comments

* adding missing bracket

* ran npm run prettier-fix to correct style issues

* resolving model/definition errors

* Revert "resolving model/definition errors"

This reverts commit fd67a4f813d923965cefad50999010820446a107.

Reverting prettier changes made to other branches

* Revert "ran npm run prettier-fix to correct style issues"

This reverts commit 3055791e00e956fff3a62e24d8313dde96daa00c.

* update

* Update DeleteWatchlist.json

* Update Watchlists.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* fixing styling issues on Delete example

* Update DeleteWatchlist.json

* Update DeleteWatchlist.json

* ran npm run prettier on DeleteWatchlist.json

* Update Watchlists.json

* reverting alias to watchlistAlias

* fixing type version

* Update Watchlists.json

* Update DeleteWatchlist.json

* fixing commas

Co-authored-by: Hassan Chawiche <hassanc@microsoft.com>
Co-authored-by: Usman Hasan <usman.hasan@microsoft.com>

royrein committed:
Done (#17843)

Co-authored-by: Roy Reinhorn <roreinho@microsoft.com>

dosegal committed:
AutomationRules 2022-01-01-preview: Add constraints (#18099)

* add contraints (alinged with the latest stable version)

akning-ms committed:
Merge branch 'main' into dev-Sentinel-2022-01-01-preview
